lang/src/spm/sol_dependency_analysers.py

```diff
@@ -0,0 +1,227 @@
+      if not len(modifier_dependencies):
+        continue
+      dependencies[contract]["modifiers"][modifier] = modifier_dependencies
+
+
+  return dependencies
+
+
+# -------- Utility funkcije za formiranje zavisnosti i formiranje json iz nase strukture --------
+
+
+def sanitize_itself(name, dependencies):
+  return list(filter(lambda dependency: name != dependency, dependencies))
+
+def sanitize_empty_values(dependencies):
+  sanitized_dependencies = {}
+  for key, value in dependencies.items():
+    if not len(dependencies[key]):
+      continue
+    sanitized_dependencies[key] = value
+
+  return sanitized_dependencies
+
+
+def find_dependencies_for_struct(sol_data, contract, struct_name):
+  struct_code = sol_data[contract]["structs"][struct_name]
+  return sanitize_empty_values({
+    "contracts": find_dependent_contracts(sol_data, struct_code),
+    "structs": sanitize_itself(struct_name, find_dependent_structs(sol_data, contract, struct_code)),
+  })
+
+def find_dependencies_for_function(sol_data, contract, function_name):
+  function_code = sol_data[contract]["functions"][function_name]
+  return sanitize_empty_values({
+    "contracts": find_dependent_contracts(sol_data, function_code),
+    "structs": find_dependent_structs(sol_data, contract, function_code),
+    "functions": sanitize_itself(function_name, find_dependent_functions(sol_data, contract, function_code)),
+    "variables": find_dependent_variables(sol_data, contract, function_code),
+    "events": find_dependent_events(sol_data, contract, function_code),
+    "modifiers": find_dependent_modifiers(sol_data, contract, function_code)
+  })
+
+def find_dependencies_for_variables(sol_data, contract, variable_name):
+  variable_code = sol_data[contract]["variables"][variable_name]
+  return sanitize_empty_values({
+    "contracts": find_dependent_contracts(sol_data, variable_code),
+    "structs": find_dependent_structs(sol_data, contract, variable_code),
+  })
+
+def find_dependencies_for_event(sol_data, contract, event_name):
+  event_code = sol_data[contract]["events"][event_name]
+  return sanitize_empty_values({
+    "contracts": find_dependent_contracts(sol_data, event_code),
+    "structs": find_dependent_structs(sol_data, contract, event_code),
+  })
+
+def find_dependencies_for_modifier(sol_data, contract, modifier_name):
+  modifier_code = sol_data[contract]["modifiers"][modifier_name]
+  return sanitize_empty_values({
+    "contracts": find_dependent_contracts(sol_data, modifier_code),
+    "structs": find_dependent_structs(sol_data, contract, modifier_code),
+    "functions": sanitize_itself(modifier_name, find_dependent_functions(sol_data, contract, modifier_code)),
+    "variables": find_dependent_variables(sol_data, contract, modifier_code),
+    "events": find_dependent_events(sol_data, contract, modifier_code)
+  })
+
+def find_dependent_contracts(sol_data, code):
+  contract_names = find_possible_contract_name(sol_data)
+  dependent_contracts = []
```

Comment:
```suggestion
  return [name for name in contract_names if re.search(fr"[^\w]{name}[^\w]", code)]
```
Use this code with regex, because `"{0} ".format(contract_name) in code` can be true if another contract's name ends with this contract's name.